File: Sum of left leaves/Sum of left leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        int sum = 0;
        preOrder(root, false, sum);
        return sum;
    }
    
    void preOrder(TreeNode *node, bool leftChild, int& sum) {
        if (node != NULL) {
            if (node->left == NULL && node->right == NULL && leftChild == true) {
                sum += node->val;
            }
            if (node->left != NULL) {
                preOrder(node->left, true, sum);
            }
            if (node->right != NULL) {
                preOrder(node->right, false, sum);
            }
        }
    }
};

File: Sum of left leaves/Sum of left leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int sum = 0;
    
    public int sumOfLeftLeaves(TreeNode root) {
        preOrder(root, false);
        return this.sum;
    }
    
    public void preOrder(TreeNode node, boolean leftChild) {
        if (node != null) {
            if (node.left == null && node.right == null && leftChild == true){ 
                this.sum += node.val;
            }
            if (node.left != null) {
                preOrder(node.left, true);
            }
            if (node.right != null) {
                preOrder(node.right, false);
            }
        }
    }
}
